tools/wake-unit/trie.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <wcl/trie.h>
+#include <wcl/xoshiro_256.h>
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <random>
+#include <string>
+#include <vector>
+
+#include "unit.h"
+
+TEST(trie_string) {
+  wcl::trie<std::string, int> test;
+  std::vector<std::string> seq = {"this", "is", "a", "test"};
```

Comment:
Maybe I'm a bit confused by the api but this looks like its a trie of `std::vector<std::string>>` instead of `std::string`.

ie
```
this -> is -> a -> test(10)
                -> different -> leaf(20)
```

we should also have a test case for the more traditional "dictionary" use case

ie
```
c -> a -> t(3)
       -> r(3) -> t(4)
       -> b(3)
```

Reply:
right, I guess I was asking: "Can we have a test for both cases"

Reply:
sure if you want, that's more or less the same as the integer tests below.